### ISSUE TITLE
Enable Object Versioning

### ISSUE THREAD

**Issue by davidcommarmond** [udx/wp-stateless#226]
Hello,
Since it's not possible to enable "Object Versioning" from the console, it would be nice to have this option via the plugin interface.

This will be particularly useful for users running WordPress in a truly stateless mode on a Kubernetes cluster for example, and want the bucket to "Backup itself", especially since they're deprecating the "Backup Entities": https://cloud.google.com/appengine/docs/standard/go/console/datastore-backing-up-restoring

So the logic would be to

1. Checking whether Object Versioning is enabled
```
curl -X GET -H "Authorization: Bearer [OAUTH2_TOKEN]" \
    "https://www.googleapis.com/storage/v1/b/[BUCKET_NAME]?fields=versioning"
```
And use result of same to show the option as enabled / disabled (To avoid conflicts if ever versioning gets changed using gsutil in parallel)

2. Depdending on state change of the selected option either
Enable versioning
```
curl -X PATCH --data-binary @[JSON_FILE_NAME].json \
    -H "Authorization: Bearer [OAUTH2_TOKEN]" \
    -H "Content-Type: application/json" \
    "https://www.googleapis.com/storage/v1/b/[BUCKET_NAME]?fields=versioning"
```

Or disable it
```
curl -X PATCH --data-binary @[JSON_FILE_NAME].json \
    -H "Authorization: Bearer [OAUTH2_TOKEN]" \
    -H "Content-Type: application/json" \
    "https://www.googleapis.com/storage/v1/b/[BUCKET_NAME]?fields=versioning"
```

And to make this solution really complete, allow lifecycle configuration as well... Although this can easily be done then via the web console instead: https://cloud.google.com/storage/docs/managing-lifecycles so can go into a nice to have list afterwards.

Thanks
